package/composition.yaml

```diff
@@ -83,3 +89,75 @@ spec:
               string:
                 type: Convert
                 convert: FromBase64
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.manifest.data.restapi-id
+          toFieldPath: status.agw.restapi-id
+          transforms:
+            - type: string
+              string:
+                type: Convert
+                convert: FromBase64
+    - name: deployment
+      base:
+        apiVersion: apigateway.aws.upbound.io/v1beta1
+        kind: Deployment
+      patches:
+        - fromFieldPath: spec.parameters.region
+          toFieldPath: spec.forProvider.region
+        - fromFieldPath: status.agw.restapi-id
+          toFieldPath: spec.forProvider.restApiId
+          policy:
+            fromFieldPath: Required
+    - name: stage
+      base:
+        apiVersion: apigateway.aws.upbound.io/v1beta1
+        kind: Stage
+        spec:
+          forProvider:
+            stageName: prod
+            deploymentIdSelector:
+              matchControllerRef: true
+      patches:
+        - fromFieldPath: spec.parameters.region
+          toFieldPath: spec.forProvider.region
+        - fromFieldPath: spec.parameters.tags
+          toFieldPath: spec.forProvider.tags
+        - fromFieldPath: status.agw.restapi-id
+          toFieldPath: spec.forProvider.restApiId
+          policy:
+            fromFieldPath: Required
+    - name: domain
```

Comment:
If we can it might be worth making this optional in some capacity (move to helmchart?)

You can have API Gateways without domains which can be executed via other services etc.

Reply:
Great point, let's figure out how to do it in the next iteration/PR. 
I included it into the static part because BasePathMapping depends on both Domain and Stage, and Stage depends on Deployment and Deployments should be triggered strictly after the core stuff ...
We'll find a way, thanks for the catch